provider/pkg/quicksight/shared-types.json

```diff
@@ -0,0 +1,445 @@
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "FieldSort"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.chart_configuration.$.sort_configuration.$.category_sort.$.column_sort",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "ColumnSort"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.actions",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "VisualCustomActions"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.chart_configuration.$.sort_configuration.$.category_sort",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "FieldSortOptions"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.chart_configuration.$.field_wells.$.bar_chart_aggregated_field_wells.$.small_multiples",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "SmallMultiplesOptions"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.chart_configuration.$.sort_configuration.$.category_items_limit",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "ItemsLimitConfiguration"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.pivot_table_visual.$.chart_configuration.$.table_options.$.cell_style",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "CellStyle"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.chart_configuration.$.category_axis",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "AxisDisplayOptions"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.kpi_visual.$.chart_configuration.$.field_wells.$.trend_groups",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "DimensionFieldSchema"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.gauge_chart_visual.$.chart_configuration.$.field_wells.$.target_values",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "MeasureFieldSchema"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.filled_map_visual.$.chart_configuration.$.field_wells.$.filled_map_aggregated_field_wells.$.values",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "MeasureFieldSchema2"
+    },
+    {
+      "location": {
+        "path": "definition.$.sheets.$.visuals.$.bar_chart_visual.$.column_hierarchies.$.predefined_hierarchy.$.columns",
+        "resource": "aws_quicksight_dashboard"
+      },
+      "typeName": "DimensionFieldSchema2"
```

Comment:
This needs another look. Equality failed to identify it with DimensionFieldSchema (no 2). Need to figure out if this is intentionally distinct and if so how to name it.